File: CS/OData.Server/API/Security/CheckPermissionController.cs
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Core;
using DevExpress.ExpressApp.DC;
using DevExpress.ExpressApp.Security;
using DevExpress.ExpressApp.Security.Authentication.ClientServer;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using Swashbuckle.AspNetCore.Annotations;

namespace OData.Server.JWT;

[Authorize]
[ApiController]
[Route("api/[controller]")]
public class CheckPermissionController : ControllerBase {
    readonly IAuthenticationTokenProvider tokenProvider;
    private readonly SecurityStrategy _security;
    private readonly IObjectSpaceFactory _securedObjectSpaceFactory;
    readonly ITypesInfo _typesInfo;

    public CheckPermissionController(ISecurityStrategyBase security, IObjectSpaceFactory securedObjectSpaceFactory, ITypesInfo typesInfo) {
        _security = (SecurityStrategy)security;
        _securedObjectSpaceFactory = securedObjectSpaceFactory;
        _typesInfo = typesInfo;
    }
    [HttpGet(nameof(CanEdit))]
    public IActionResult CanEdit(string typeName, string objectKey) {
        var type = _typesInfo.FindTypeInfo(typeName)?.Type;
        using(var objectSpace = _securedObjectSpaceFactory.CreateObjectSpace(type)) {
            var key = objectSpace.GetObjectKey(type, objectKey);
            return Ok(_security.CanWrite(type, objectSpace, key));
        }
    }
}

File: VB/OData.Server/API/Security/CheckPermissionController.vb
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Core
Imports DevExpress.ExpressApp.DC
Imports DevExpress.ExpressApp.Security
Imports DevExpress.ExpressApp.Security.Authentication.ClientServer
Imports Microsoft.AspNetCore.Authorization
Imports Microsoft.AspNetCore.Mvc
Imports Swashbuckle.AspNetCore.Annotations

Namespace OData.Server.JWT

    <Authorize>
    <ApiController>
    <Route("api/[controller]")>
    Public Class CheckPermissionController
        Inherits ControllerBase

        Private ReadOnly tokenProvider As IAuthenticationTokenProvider

        Private ReadOnly _security As SecurityStrategy

        Private ReadOnly _securedObjectSpaceFactory As IObjectSpaceFactory

        Private ReadOnly _typesInfo As ITypesInfo

        Public Sub New(ByVal security As ISecurityStrategyBase, ByVal securedObjectSpaceFactory As IObjectSpaceFactory, ByVal typesInfo As ITypesInfo)
            _security = CType(security, SecurityStrategy)
            _securedObjectSpaceFactory = securedObjectSpaceFactory
            _typesInfo = typesInfo
        End Sub

        <HttpGet(NameOf(CanEdit))>
        Public Function CanEdit(ByVal typeName As String, ByVal objectKey As String) As IActionResult
            Dim type = _typesInfo.FindTypeInfo(typeName)?.Type
            Using objectSpace = _securedObjectSpaceFactory.CreateObjectSpace(type)
                Dim key = objectSpace.GetObjectKey(type, objectKey)
                Return Ok(_security.CanWrite(type, objectSpace, key))
            End Using
        End Function
    End Class
End Namespace
